packages/rocketchat-api/server/v1/spotlight.js

```diff
@@ -0,0 +1,19 @@
+RocketChat.API.v1.addRoute('spotlight', { authRequired: true }, {
+	get() {
+		check(this.queryParams, {
+			query: String
+		});
+
+		const { query } = this.queryParams;
+
+		let result;
+		Meteor.runAsUser(this.userId, () =>
+			result = Meteor.call('spotlight', query, null, {
+				rooms: true,
+				users: true
+			})
+		);
```

Comment:
@RocketChat/core For some reason, this method ALWAYS returns `rooms` empty. I tried debugging it already and it's entering the part of the code that queries per rooms, so I don't know what's going on. Here's the call example I'm using:

```shell
curl -X GET -H "X-Auth-Token: 6eWs4ECxUITYYoZ_bAYK5OE2srkxAepQqQA1cGGv3jK" -H "X-User-Id: hvzu8z6mHFigiXy6Y" http://localhost:3000/api/v1/spotlight?query=t
```

And I do have a room with the name of `general` and `testing-public-channel`. The users I'm trying also have all the permissions required to search by rooms: `view-outside-room` and `view-c-room`. Does someone have any idea what's going on?

Reply:
The query must be a valid json object.

Reply:
@graywolf336 What do you mean? It works for users... do you have an example of how it should be?

Reply:
Never mind, I thought you was doing something else. Ignore my comment.

Reply:
@rafaelks If you call it via DDP it returns data for rooms?

Reply:
@rodrigok How can I do that?

Reply:
Exactly as you wrote for the server side call, but from you browser (logged in)
```
Meteor.call('spotlight', 't', null, {
				rooms: true,
				users: true
			})
```

Reply:
Per our chat, the API is correct & it's not returning the objects because user already joined the rooms.